### DISCUSSION TITLE
Controlled Dialog and onOpenChange inconsistency

### DISCUSSION THREAD

**Discussion by dannylin108** [corvudev/corvu#47] [Q&A] [answered]
If I don't use a Dialog.Trigger, but use `open` and `onOpenChange` props for Dialog.Root, I'm noticing that `onOpenChange` is called only when I close the  dialog, but not when I open it. Is that intentional?

In persistent dialog pattern that opening `onOpenChange` could be useful for state initialization...

**Comment by GiyoMoon** [comment, marked as answer]
`onOpenChange` only gets called when corvu changes the open state internally. The purpose of this callback is to allow you to reflect state changes to your signal, if you directly edit your signal you don't have to be notified because your signal is already up to date. Normally you'd put `setOpen` inside onOpenChange to sync it:
`<Dialog open={open()} onOpenChange={setOpen} />`
